tests/services/test_fpc_balance_service.py

```diff
@@ -0,0 +1,166 @@
+from decimal import Decimal
+
+from parameterized import parameterized
+
+from arbeitszeit.fpc_balance import PublicFundService
+from arbeitszeit.records import ProductionCosts
+from tests.data_generators import TransactionGenerator
+from tests.use_cases.base_test_case import BaseTestCase
+
+
+class PublicFundServiceCalculationTests(BaseTestCase):
+    def setUp(self) -> None:
+        super().setUp()
+        self.service = self.injector.get(PublicFundService)
+        self.transaction_generator = self.injector.get(TransactionGenerator)
+
+    def test_fpc_balance_is_zero_if_no_public_plans_and_no_transactions(self) -> None:
+        fpc_balance = self.service.calculate_fpc_balance()
+        self.assertEqual(fpc_balance, 0)
+
+    @parameterized.expand(
+        [
+            (Decimal(10), Decimal(0.95), 10, Decimal(5)),
+            (Decimal(1), Decimal(1), 10, Decimal(0)),
+            (Decimal(28), Decimal(0.15), 100, Decimal(2380)),
+            (Decimal(28), Decimal(0.9), 100, Decimal(280)),
+            (Decimal(2), Decimal(-0.5), 10, Decimal(30)),
+        ]
+    )
+    def test_fpc_balance_if_no_public_plans(
```

Comment:
The names of this and the following tests could be more expressive.

Reply:
done